packages/app/cypress/e2e/runner/sessions.ui.cy.ts

```diff
@@ -315,34 +313,375 @@ describe('runner/cypress sessions.ui.spec', {
+                hasCodeFrame,
+              })
+            })
+
+            cy.get('@example_test')
+            .find('.attempt-error-region')
+            .should('not.exist')
+          })
         })
       })
 
-      // has error
-      cy.get('@setup_failed').contains('This error occurred while creating session. Because the session setup failed, we failed the test.')
+      describe('failed to recreated session', () => {
+        before(() => {
+          loadSpec({
+            projectName: 'session-and-origin-e2e-specs',
+            filePath: 'session/errors.cy.js',
+            passCount: 0,
+            failCount: 7,
+            setup () {
+              cy.window().then((win) => {
+                // @ts-ignore
+                return win.CYPRESS_TEST_DATA = {
+                  restoreSessionWithValidationFailure: true,
+                  successfullyRecreatedSession: false,
+                }
+              })
+            },
+          })
+        })
+
+        it('setup has failing command', () => {
+          cy.contains('.test', 'setup - has failing command').as('example_test')
+
+          cy.get('@example_test').within(() => {
+            assertRecreatedSession({
+              testAlias: '@example_test',
+              validationErrMessage: 'callback yielded false',
+              commandPassed: false,
+              successfullyRecreatedSession: false,
+            })
+          })
+
+          const recreatedErrPostfix = 'This error occurred while creating the session. Because the session setup failed, we failed the test.'
+
+          cy.get('@example_test')
+          .find('.attempt-error-region')
+          .contains('Expected to find element')
+          .contains(recreatedErrPostfix)
+        })
+
+        describe('failed validation', () => {
+          [
+            {
+              testCase: 'has failing Cypress command',
+              systemTestTitle: 'validate - has failing Cypress command',
+              errMessage: 'failed because it requires a DOM element',
+            },
+            {
+              testCase: 'command yields false',
+              systemTestTitle: 'validate - command yields false',
+              errMessage: 'callback yielded false.',
+            },
+            {
+              testCase: 'has multiple commands and yields false',
+              systemTestTitle: 'validate - has multiple commands and yields false',
+              errMessage: 'callback yielded false.',
+            },
+            {
+              testCase: 'rejects with false',
+              systemTestTitle: 'validate - rejects with false',
+              errMessage: 'rejected with false.',
+            },
+            {
+              testCase: 'promise resolved false',
+              systemTestTitle: 'validate - promise resolves false',
+              errMessage: 'promise resolved false.',
+            },
+            {
+              testCase: 'throws an error',
```

Comment:
Looks like I have lost the stack trace.. But still better

**Before**

https://user-images.githubusercontent.com/14099737/199747622-d31788fb-92c7-4c29-91a3-fdfb2acbec21.mov

**After**

https://user-images.githubusercontent.com/14099737/199747581-3f6d600c-c0e7-4067-b571-2033fc582e9a.mov